// @prisma/client/models/contacts_contact_rows.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file exports the `contacts_contact_rows` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/client"
import type * as $Enums from "../enums.ts"
import type * as Prisma from "../internal/prismaNamespace.ts"

/**
 * Model contacts_contact_rows
 * 
 */
export type contacts_contact_rowsModel = runtime.Types.Result.DefaultSelection<Prisma.$contacts_contact_rowsPayload>

export type AggregateContacts_contact_rows = {
  _count: Contacts_contact_rowsCountAggregateOutputType | null
  _min: Contacts_contact_rowsMinAggregateOutputType | null
  _max: Contacts_contact_rowsMaxAggregateOutputType | null
}

export type Contacts_contact_rowsMinAggregateOutputType = {
  id: string | null
  first_name: string | null
  last_name: string | null
  email: string | null
  phone: string | null
  title: string | null
  email_type: string | null
  contact_form_url: string | null
  created_at: Date | null
  updated_at: Date | null
  department: string | null
  agency_id: string | null
  firm_id: string | null
}

export type Contacts_contact_rowsMaxAggregateOutputType = {
  id: string | null
  first_name: string | null
  last_name: string | null
  email: string | null
  phone: string | null
  title: string | null
  email_type: string | null
  contact_form_url: string | null
  created_at: Date | null
  updated_at: Date | null
  department: string | null
  agency_id: string | null
  firm_id: string | null
}

export type Contacts_contact_rowsCountAggregateOutputType = {
  id: number
  first_name: number
  last_name: number
  email: number
  phone: number
  title: number
  email_type: number
  contact_form_url: number
  created_at: number
  updated_at: number
  department: number
  agency_id: number
  firm_id: number
  _all: number
}


export type Contacts_contact_rowsMinAggregateInputType = {
  id?: true
  first_name?: true
  last_name?: true
  email?: true
  phone?: true
  title?: true
  email_type?: true
  contact_form_url?: true
  created_at?: true
  updated_at?: true
  department?: true
  agency_id?: true
  firm_id?: true
}

export type Contacts_contact_rowsMaxAggregateInputType = {
  id?: true
  first_name?: true
  last_name?: true
  email?: true
  phone?: true
  title?: true
  email_type?: true
  contact_form_url?: true
  created_at?: true
  updated_at?: true
  department?: true
  agency_id?: true
  firm_id?: true
}

export type Contacts_contact_rowsCountAggregateInputType = {
  id?: true
  first_name?: true
  last_name?: true
  email?: true
  phone?: true
  title?: true
  email_type?: true
  contact_form_url?: true
  created_at?: true
  updated_at?: true
  department?: true
  agency_id?: true
  firm_id?: true
  _all?: true
}

export type Contacts_contact_rowsAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which contacts_contact_rows to aggregate.
   */
  where?: Prisma.contacts_contact_rowsWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of contacts_contact_rows to fetch.
   */
  orderBy?: Prisma.contacts_contact_rowsOrderByWithRelationInput | Prisma.contacts_contact_rowsOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.contacts_contact_rowsWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` contacts_contact_rows from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` contacts_contact_rows.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned contacts_contact_rows
  **/
  _count?: true | Contacts_contact_rowsCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: Contacts_contact_rowsMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: Contacts_contact_rowsMaxAggregateInputType
}

export type GetContacts_contact_rowsAggregateType<T extends Contacts_contact_rowsAggregateArgs> = {
      [P in keyof T & keyof AggregateContacts_contact_rows]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateContacts_contact_rows[P]>
    : Prisma.GetScalarType<T[P], AggregateContacts_contact_rows[P]>
}




export type contacts_contact_rowsGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.contacts_contact_rowsWhereInput
  orderBy?: Prisma.contacts_contact_rowsOrderByWithAggregationInput | Prisma.contacts_contact_rowsOrderByWithAggregationInput[]
  by: Prisma.Contacts_contact_rowsScalarFieldEnum[] | Prisma.Contacts_contact_rowsScalarFieldEnum
  having?: Prisma.contacts_contact_rowsScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: Contacts_contact_rowsCountAggregateInputType | true
  _min?: Contacts_contact_rowsMinAggregateInputType
  _max?: Contacts_contact_rowsMaxAggregateInputType
}

export type Contacts_contact_rowsGroupByOutputType = {
  id: string
  first_name: string | null
  last_name: string | null
  email: string | null
  phone: string | null
  title: string | null
  email_type: string | null
  contact_form_url: string | null
  created_at: Date | null
  updated_at: Date | null
  department: string | null
  agency_id: string | null
  firm_id: string | null
  _count: Contacts_contact_rowsCountAggregateOutputType | null
  _min: Contacts_contact_rowsMinAggregateOutputType | null
  _max: Contacts_contact_rowsMaxAggregateOutputType | null
}

type GetContacts_contact_rowsGroupByPayload<T extends contacts_contact_rowsGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<Contacts_contact_rowsGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof Contacts_contact_rowsGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], Contacts_contact_rowsGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], Contacts_contact_rowsGroupByOutputType[P]>
      }
    >
  >



export type contacts_contact_rowsWhereInput = {
  AND?: Prisma.contacts_contact_rowsWhereInput | Prisma.contacts_contact_rowsWhereInput[]
  OR?: Prisma.contacts_contact_rowsWhereInput[]
  NOT?: Prisma.contacts_contact_rowsWhereInput | Prisma.contacts_contact_rowsWhereInput[]
  id?: Prisma.StringFilter<"contacts_contact_rows"> | string
  first_name?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  last_name?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  email?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  phone?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  title?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  email_type?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  contact_form_url?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  created_at?: Prisma.DateTimeNullableFilter<"contacts_contact_rows"> | Date | string | null
  updated_at?: Prisma.DateTimeNullableFilter<"contacts_contact_rows"> | Date | string | null
  department?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  agency_id?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  firm_id?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  agency?: Prisma.XOR<Prisma.Agencies_agency_rowsNullableScalarRelationFilter, Prisma.agencies_agency_rowsWhereInput> | null
}

export type contacts_contact_rowsOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  first_name?: Prisma.SortOrderInput | Prisma.SortOrder
  last_name?: Prisma.SortOrderInput | Prisma.SortOrder
  email?: Prisma.SortOrderInput | Prisma.SortOrder
  phone?: Prisma.SortOrderInput | Prisma.SortOrder
  title?: Prisma.SortOrderInput | Prisma.SortOrder
  email_type?: Prisma.SortOrderInput | Prisma.SortOrder
  contact_form_url?: Prisma.SortOrderInput | Prisma.SortOrder
  created_at?: Prisma.SortOrderInput | Prisma.SortOrder
  updated_at?: Prisma.SortOrderInput | Prisma.SortOrder
  department?: Prisma.SortOrderInput | Prisma.SortOrder
  agency_id?: Prisma.SortOrderInput | Prisma.SortOrder
  firm_id?: Prisma.SortOrderInput | Prisma.SortOrder
  agency?: Prisma.agencies_agency_rowsOrderByWithRelationInput
}

export type contacts_contact_rowsWhereUniqueInput = Prisma.AtLeast<{
  id?: string
  AND?: Prisma.contacts_contact_rowsWhereInput | Prisma.contacts_contact_rowsWhereInput[]
  OR?: Prisma.contacts_contact_rowsWhereInput[]
  NOT?: Prisma.contacts_contact_rowsWhereInput | Prisma.contacts_contact_rowsWhereInput[]
  first_name?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  last_name?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  email?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  phone?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  title?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  email_type?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  contact_form_url?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  created_at?: Prisma.DateTimeNullableFilter<"contacts_contact_rows"> | Date | string | null
  updated_at?: Prisma.DateTimeNullableFilter<"contacts_contact_rows"> | Date | string | null
  department?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  agency_id?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  firm_id?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  agency?: Prisma.XOR<Prisma.Agencies_agency_rowsNullableScalarRelationFilter, Prisma.agencies_agency_rowsWhereInput> | null
}, "id">

export type contacts_contact_rowsOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  first_name?: Prisma.SortOrderInput | Prisma.SortOrder
  last_name?: Prisma.SortOrderInput | Prisma.SortOrder
  email?: Prisma.SortOrderInput | Prisma.SortOrder
  phone?: Prisma.SortOrderInput | Prisma.SortOrder
  title?: Prisma.SortOrderInput | Prisma.SortOrder
  email_type?: Prisma.SortOrderInput | Prisma.SortOrder
  contact_form_url?: Prisma.SortOrderInput | Prisma.SortOrder
  created_at?: Prisma.SortOrderInput | Prisma.SortOrder
  updated_at?: Prisma.SortOrderInput | Prisma.SortOrder
  department?: Prisma.SortOrderInput | Prisma.SortOrder
  agency_id?: Prisma.SortOrderInput | Prisma.SortOrder
  firm_id?: Prisma.SortOrderInput | Prisma.SortOrder
  _count?: Prisma.contacts_contact_rowsCountOrderByAggregateInput
  _max?: Prisma.contacts_contact_rowsMaxOrderByAggregateInput
  _min?: Prisma.contacts_contact_rowsMinOrderByAggregateInput
}

export type contacts_contact_rowsScalarWhereWithAggregatesInput = {
  AND?: Prisma.contacts_contact_rowsScalarWhereWithAggregatesInput | Prisma.contacts_contact_rowsScalarWhereWithAggregatesInput[]
  OR?: Prisma.contacts_contact_rowsScalarWhereWithAggregatesInput[]
  NOT?: Prisma.contacts_contact_rowsScalarWhereWithAggregatesInput | Prisma.contacts_contact_rowsScalarWhereWithAggregatesInput[]
  id?: Prisma.StringWithAggregatesFilter<"contacts_contact_rows"> | string
  first_name?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
  last_name?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
  email?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
  phone?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
  title?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
  email_type?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
  contact_form_url?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
  created_at?: Prisma.DateTimeNullableWithAggregatesFilter<"contacts_contact_rows"> | Date | string | null
  updated_at?: Prisma.DateTimeNullableWithAggregatesFilter<"contacts_contact_rows"> | Date | string | null
  department?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
  agency_id?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
  firm_id?: Prisma.StringNullableWithAggregatesFilter<"contacts_contact_rows"> | string | null
}

export type contacts_contact_rowsCreateInput = {
  id?: string
  first_name?: string | null
  last_name?: string | null
  email?: string | null
  phone?: string | null
  title?: string | null
  email_type?: string | null
  contact_form_url?: string | null
  created_at?: Date | string | null
  updated_at?: Date | string | null
  department?: string | null
  firm_id?: string | null
  agency?: Prisma.agencies_agency_rowsCreateNestedOneWithoutContactsInput
}

export type contacts_contact_rowsUncheckedCreateInput = {
  id?: string
  first_name?: string | null
  last_name?: string | null
  email?: string | null
  phone?: string | null
  title?: string | null
  email_type?: string | null
  contact_form_url?: string | null
  created_at?: Date | string | null
  updated_at?: Date | string | null
  department?: string | null
  agency_id?: string | null
  firm_id?: string | null
}

export type contacts_contact_rowsUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  first_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  last_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  phone?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email_type?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  contact_form_url?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  created_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  updated_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  department?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  firm_id?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  agency?: Prisma.agencies_agency_rowsUpdateOneWithoutContactsNestedInput
}

export type contacts_contact_rowsUncheckedUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  first_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  last_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  phone?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email_type?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  contact_form_url?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  created_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  updated_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  department?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  agency_id?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  firm_id?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type contacts_contact_rowsCreateManyInput = {
  id?: string
  first_name?: string | null
  last_name?: string | null
  email?: string | null
  phone?: string | null
  title?: string | null
  email_type?: string | null
  contact_form_url?: string | null
  created_at?: Date | string | null
  updated_at?: Date | string | null
  department?: string | null
  agency_id?: string | null
  firm_id?: string | null
}

export type contacts_contact_rowsUpdateManyMutationInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  first_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  last_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  phone?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email_type?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  contact_form_url?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  created_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  updated_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  department?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  firm_id?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type contacts_contact_rowsUncheckedUpdateManyInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  first_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  last_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  phone?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email_type?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  contact_form_url?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  created_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  updated_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  department?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  agency_id?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  firm_id?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type Contacts_contact_rowsListRelationFilter = {
  every?: Prisma.contacts_contact_rowsWhereInput
  some?: Prisma.contacts_contact_rowsWhereInput
  none?: Prisma.contacts_contact_rowsWhereInput
}

export type contacts_contact_rowsOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type contacts_contact_rowsCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  first_name?: Prisma.SortOrder
  last_name?: Prisma.SortOrder
  email?: Prisma.SortOrder
  phone?: Prisma.SortOrder
  title?: Prisma.SortOrder
  email_type?: Prisma.SortOrder
  contact_form_url?: Prisma.SortOrder
  created_at?: Prisma.SortOrder
  updated_at?: Prisma.SortOrder
  department?: Prisma.SortOrder
  agency_id?: Prisma.SortOrder
  firm_id?: Prisma.SortOrder
}

export type contacts_contact_rowsMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  first_name?: Prisma.SortOrder
  last_name?: Prisma.SortOrder
  email?: Prisma.SortOrder
  phone?: Prisma.SortOrder
  title?: Prisma.SortOrder
  email_type?: Prisma.SortOrder
  contact_form_url?: Prisma.SortOrder
  created_at?: Prisma.SortOrder
  updated_at?: Prisma.SortOrder
  department?: Prisma.SortOrder
  agency_id?: Prisma.SortOrder
  firm_id?: Prisma.SortOrder
}

export type contacts_contact_rowsMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  first_name?: Prisma.SortOrder
  last_name?: Prisma.SortOrder
  email?: Prisma.SortOrder
  phone?: Prisma.SortOrder
  title?: Prisma.SortOrder
  email_type?: Prisma.SortOrder
  contact_form_url?: Prisma.SortOrder
  created_at?: Prisma.SortOrder
  updated_at?: Prisma.SortOrder
  department?: Prisma.SortOrder
  agency_id?: Prisma.SortOrder
  firm_id?: Prisma.SortOrder
}

export type contacts_contact_rowsCreateNestedManyWithoutAgencyInput = {
  create?: Prisma.XOR<Prisma.contacts_contact_rowsCreateWithoutAgencyInput, Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput> | Prisma.contacts_contact_rowsCreateWithoutAgencyInput[] | Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput[]
  connectOrCreate?: Prisma.contacts_contact_rowsCreateOrConnectWithoutAgencyInput | Prisma.contacts_contact_rowsCreateOrConnectWithoutAgencyInput[]
  createMany?: Prisma.contacts_contact_rowsCreateManyAgencyInputEnvelope
  connect?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
}

export type contacts_contact_rowsUncheckedCreateNestedManyWithoutAgencyInput = {
  create?: Prisma.XOR<Prisma.contacts_contact_rowsCreateWithoutAgencyInput, Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput> | Prisma.contacts_contact_rowsCreateWithoutAgencyInput[] | Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput[]
  connectOrCreate?: Prisma.contacts_contact_rowsCreateOrConnectWithoutAgencyInput | Prisma.contacts_contact_rowsCreateOrConnectWithoutAgencyInput[]
  createMany?: Prisma.contacts_contact_rowsCreateManyAgencyInputEnvelope
  connect?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
}

export type contacts_contact_rowsUpdateManyWithoutAgencyNestedInput = {
  create?: Prisma.XOR<Prisma.contacts_contact_rowsCreateWithoutAgencyInput, Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput> | Prisma.contacts_contact_rowsCreateWithoutAgencyInput[] | Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput[]
  connectOrCreate?: Prisma.contacts_contact_rowsCreateOrConnectWithoutAgencyInput | Prisma.contacts_contact_rowsCreateOrConnectWithoutAgencyInput[]
  upsert?: Prisma.contacts_contact_rowsUpsertWithWhereUniqueWithoutAgencyInput | Prisma.contacts_contact_rowsUpsertWithWhereUniqueWithoutAgencyInput[]
  createMany?: Prisma.contacts_contact_rowsCreateManyAgencyInputEnvelope
  set?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
  disconnect?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
  delete?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
  connect?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
  update?: Prisma.contacts_contact_rowsUpdateWithWhereUniqueWithoutAgencyInput | Prisma.contacts_contact_rowsUpdateWithWhereUniqueWithoutAgencyInput[]
  updateMany?: Prisma.contacts_contact_rowsUpdateManyWithWhereWithoutAgencyInput | Prisma.contacts_contact_rowsUpdateManyWithWhereWithoutAgencyInput[]
  deleteMany?: Prisma.contacts_contact_rowsScalarWhereInput | Prisma.contacts_contact_rowsScalarWhereInput[]
}

export type contacts_contact_rowsUncheckedUpdateManyWithoutAgencyNestedInput = {
  create?: Prisma.XOR<Prisma.contacts_contact_rowsCreateWithoutAgencyInput, Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput> | Prisma.contacts_contact_rowsCreateWithoutAgencyInput[] | Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput[]
  connectOrCreate?: Prisma.contacts_contact_rowsCreateOrConnectWithoutAgencyInput | Prisma.contacts_contact_rowsCreateOrConnectWithoutAgencyInput[]
  upsert?: Prisma.contacts_contact_rowsUpsertWithWhereUniqueWithoutAgencyInput | Prisma.contacts_contact_rowsUpsertWithWhereUniqueWithoutAgencyInput[]
  createMany?: Prisma.contacts_contact_rowsCreateManyAgencyInputEnvelope
  set?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
  disconnect?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
  delete?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
  connect?: Prisma.contacts_contact_rowsWhereUniqueInput | Prisma.contacts_contact_rowsWhereUniqueInput[]
  update?: Prisma.contacts_contact_rowsUpdateWithWhereUniqueWithoutAgencyInput | Prisma.contacts_contact_rowsUpdateWithWhereUniqueWithoutAgencyInput[]
  updateMany?: Prisma.contacts_contact_rowsUpdateManyWithWhereWithoutAgencyInput | Prisma.contacts_contact_rowsUpdateManyWithWhereWithoutAgencyInput[]
  deleteMany?: Prisma.contacts_contact_rowsScalarWhereInput | Prisma.contacts_contact_rowsScalarWhereInput[]
}

export type contacts_contact_rowsCreateWithoutAgencyInput = {
  id?: string
  first_name?: string | null
  last_name?: string | null
  email?: string | null
  phone?: string | null
  title?: string | null
  email_type?: string | null
  contact_form_url?: string | null
  created_at?: Date | string | null
  updated_at?: Date | string | null
  department?: string | null
  firm_id?: string | null
}

export type contacts_contact_rowsUncheckedCreateWithoutAgencyInput = {
  id?: string
  first_name?: string | null
  last_name?: string | null
  email?: string | null
  phone?: string | null
  title?: string | null
  email_type?: string | null
  contact_form_url?: string | null
  created_at?: Date | string | null
  updated_at?: Date | string | null
  department?: string | null
  firm_id?: string | null
}

export type contacts_contact_rowsCreateOrConnectWithoutAgencyInput = {
  where: Prisma.contacts_contact_rowsWhereUniqueInput
  create: Prisma.XOR<Prisma.contacts_contact_rowsCreateWithoutAgencyInput, Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput>
}

export type contacts_contact_rowsCreateManyAgencyInputEnvelope = {
  data: Prisma.contacts_contact_rowsCreateManyAgencyInput | Prisma.contacts_contact_rowsCreateManyAgencyInput[]
  skipDuplicates?: boolean
}

export type contacts_contact_rowsUpsertWithWhereUniqueWithoutAgencyInput = {
  where: Prisma.contacts_contact_rowsWhereUniqueInput
  update: Prisma.XOR<Prisma.contacts_contact_rowsUpdateWithoutAgencyInput, Prisma.contacts_contact_rowsUncheckedUpdateWithoutAgencyInput>
  create: Prisma.XOR<Prisma.contacts_contact_rowsCreateWithoutAgencyInput, Prisma.contacts_contact_rowsUncheckedCreateWithoutAgencyInput>
}

export type contacts_contact_rowsUpdateWithWhereUniqueWithoutAgencyInput = {
  where: Prisma.contacts_contact_rowsWhereUniqueInput
  data: Prisma.XOR<Prisma.contacts_contact_rowsUpdateWithoutAgencyInput, Prisma.contacts_contact_rowsUncheckedUpdateWithoutAgencyInput>
}

export type contacts_contact_rowsUpdateManyWithWhereWithoutAgencyInput = {
  where: Prisma.contacts_contact_rowsScalarWhereInput
  data: Prisma.XOR<Prisma.contacts_contact_rowsUpdateManyMutationInput, Prisma.contacts_contact_rowsUncheckedUpdateManyWithoutAgencyInput>
}

export type contacts_contact_rowsScalarWhereInput = {
  AND?: Prisma.contacts_contact_rowsScalarWhereInput | Prisma.contacts_contact_rowsScalarWhereInput[]
  OR?: Prisma.contacts_contact_rowsScalarWhereInput[]
  NOT?: Prisma.contacts_contact_rowsScalarWhereInput | Prisma.contacts_contact_rowsScalarWhereInput[]
  id?: Prisma.StringFilter<"contacts_contact_rows"> | string
  first_name?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  last_name?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  email?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  phone?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  title?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  email_type?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  contact_form_url?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  created_at?: Prisma.DateTimeNullableFilter<"contacts_contact_rows"> | Date | string | null
  updated_at?: Prisma.DateTimeNullableFilter<"contacts_contact_rows"> | Date | string | null
  department?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  agency_id?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
  firm_id?: Prisma.StringNullableFilter<"contacts_contact_rows"> | string | null
}

export type contacts_contact_rowsCreateManyAgencyInput = {
  id?: string
  first_name?: string | null
  last_name?: string | null
  email?: string | null
  phone?: string | null
  title?: string | null
  email_type?: string | null
  contact_form_url?: string | null
  created_at?: Date | string | null
  updated_at?: Date | string | null
  department?: string | null
  firm_id?: string | null
}

export type contacts_contact_rowsUpdateWithoutAgencyInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  first_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  last_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  phone?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email_type?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  contact_form_url?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  created_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  updated_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  department?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  firm_id?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type contacts_contact_rowsUncheckedUpdateWithoutAgencyInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  first_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  last_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  phone?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email_type?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  contact_form_url?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  created_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  updated_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  department?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  firm_id?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type contacts_contact_rowsUncheckedUpdateManyWithoutAgencyInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  first_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  last_name?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  phone?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  title?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  email_type?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  contact_form_url?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  created_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  updated_at?: Prisma.NullableDateTimeFieldUpdateOperationsInput | Date | string | null
  department?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  firm_id?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}



export type contacts_contact_rowsSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  first_name?: boolean
  last_name?: boolean
  email?: boolean
  phone?: boolean
  title?: boolean
  email_type?: boolean
  contact_form_url?: boolean
  created_at?: boolean
  updated_at?: boolean
  department?: boolean
  agency_id?: boolean
  firm_id?: boolean
  agency?: boolean | Prisma.contacts_contact_rows$agencyArgs<ExtArgs>
}, ExtArgs["result"]["contacts_contact_rows"]>

export type contacts_contact_rowsSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  first_name?: boolean
  last_name?: boolean
  email?: boolean
  phone?: boolean
  title?: boolean
  email_type?: boolean
  contact_form_url?: boolean
  created_at?: boolean
  updated_at?: boolean
  department?: boolean
  agency_id?: boolean
  firm_id?: boolean
  agency?: boolean | Prisma.contacts_contact_rows$agencyArgs<ExtArgs>
}, ExtArgs["result"]["contacts_contact_rows"]>

export type contacts_contact_rowsSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  first_name?: boolean
  last_name?: boolean
  email?: boolean
  phone?: boolean
  title?: boolean
  email_type?: boolean
  contact_form_url?: boolean
  created_at?: boolean
  updated_at?: boolean
  department?: boolean
  agency_id?: boolean
  firm_id?: boolean
  agency?: boolean | Prisma.contacts_contact_rows$agencyArgs<ExtArgs>
}, ExtArgs["result"]["contacts_contact_rows"]>

export type contacts_contact_rowsSelectScalar = {
  id?: boolean
  first_name?: boolean
  last_name?: boolean
  email?: boolean
  phone?: boolean
  title?: boolean
  email_type?: boolean
  contact_form_url?: boolean
  created_at?: boolean
  updated_at?: boolean
  department?: boolean
  agency_id?: boolean
  firm_id?: boolean
}

export type contacts_contact_rowsOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "first_name" | "last_name" | "email" | "phone" | "title" | "email_type" | "contact_form_url" | "created_at" | "updated_at" | "department" | "agency_id" | "firm_id", ExtArgs["result"]["contacts_contact_rows"]>
export type contacts_contact_rowsInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  agency?: boolean | Prisma.contacts_contact_rows$agencyArgs<ExtArgs>
}
export type contacts_contact_rowsIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  agency?: boolean | Prisma.contacts_contact_rows$agencyArgs<ExtArgs>
}
export type contacts_contact_rowsIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  agency?: boolean | Prisma.contacts_contact_rows$agencyArgs<ExtArgs>
}

export type $contacts_contact_rowsPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "contacts_contact_rows"
  objects: {
    agency: Prisma.$agencies_agency_rowsPayload<ExtArgs> | null
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: string
    first_name: string | null
    last_name: string | null
    email: string | null
    phone: string | null
    title: string | null
    email_type: string | null
    contact_form_url: string | null
    created_at: Date | null
    updated_at: Date | null
    department: string | null
    agency_id: string | null
    firm_id: string | null
  }, ExtArgs["result"]["contacts_contact_rows"]>
  composites: {}
}

export type contacts_contact_rowsGetPayload<S extends boolean | null | undefined | contacts_contact_rowsDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload, S>

export type contacts_contact_rowsCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<contacts_contact_rowsFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: Contacts_contact_rowsCountAggregateInputType | true
  }

export interface contacts_contact_rowsDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['contacts_contact_rows'], meta: { name: 'contacts_contact_rows' } }
  /**
   * Find zero or one Contacts_contact_rows that matches the filter.
   * @param {contacts_contact_rowsFindUniqueArgs} args - Arguments to find a Contacts_contact_rows
   * @example
   * // Get one Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends contacts_contact_rowsFindUniqueArgs>(args: Prisma.SelectSubset<T, contacts_contact_rowsFindUniqueArgs<ExtArgs>>): Prisma.Prisma__contacts_contact_rowsClient<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one Contacts_contact_rows that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {contacts_contact_rowsFindUniqueOrThrowArgs} args - Arguments to find a Contacts_contact_rows
   * @example
   * // Get one Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends contacts_contact_rowsFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, contacts_contact_rowsFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__contacts_contact_rowsClient<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Contacts_contact_rows that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {contacts_contact_rowsFindFirstArgs} args - Arguments to find a Contacts_contact_rows
   * @example
   * // Get one Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends contacts_contact_rowsFindFirstArgs>(args?: Prisma.SelectSubset<T, contacts_contact_rowsFindFirstArgs<ExtArgs>>): Prisma.Prisma__contacts_contact_rowsClient<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Contacts_contact_rows that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {contacts_contact_rowsFindFirstOrThrowArgs} args - Arguments to find a Contacts_contact_rows
   * @example
   * // Get one Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends contacts_contact_rowsFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, contacts_contact_rowsFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__contacts_contact_rowsClient<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more Contacts_contact_rows that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {contacts_contact_rowsFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.findMany()
   * 
   * // Get first 10 Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const contacts_contact_rowsWithIdOnly = await prisma.contacts_contact_rows.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends contacts_contact_rowsFindManyArgs>(args?: Prisma.SelectSubset<T, contacts_contact_rowsFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a Contacts_contact_rows.
   * @param {contacts_contact_rowsCreateArgs} args - Arguments to create a Contacts_contact_rows.
   * @example
   * // Create one Contacts_contact_rows
   * const Contacts_contact_rows = await prisma.contacts_contact_rows.create({
   *   data: {
   *     // ... data to create a Contacts_contact_rows
   *   }
   * })
   * 
   */
  create<T extends contacts_contact_rowsCreateArgs>(args: Prisma.SelectSubset<T, contacts_contact_rowsCreateArgs<ExtArgs>>): Prisma.Prisma__contacts_contact_rowsClient<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many Contacts_contact_rows.
   * @param {contacts_contact_rowsCreateManyArgs} args - Arguments to create many Contacts_contact_rows.
   * @example
   * // Create many Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends contacts_contact_rowsCreateManyArgs>(args?: Prisma.SelectSubset<T, contacts_contact_rowsCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many Contacts_contact_rows and returns the data saved in the database.
   * @param {contacts_contact_rowsCreateManyAndReturnArgs} args - Arguments to create many Contacts_contact_rows.
   * @example
   * // Create many Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many Contacts_contact_rows and only return the `id`
   * const contacts_contact_rowsWithIdOnly = await prisma.contacts_contact_rows.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends contacts_contact_rowsCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, contacts_contact_rowsCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a Contacts_contact_rows.
   * @param {contacts_contact_rowsDeleteArgs} args - Arguments to delete one Contacts_contact_rows.
   * @example
   * // Delete one Contacts_contact_rows
   * const Contacts_contact_rows = await prisma.contacts_contact_rows.delete({
   *   where: {
   *     // ... filter to delete one Contacts_contact_rows
   *   }
   * })
   * 
   */
  delete<T extends contacts_contact_rowsDeleteArgs>(args: Prisma.SelectSubset<T, contacts_contact_rowsDeleteArgs<ExtArgs>>): Prisma.Prisma__contacts_contact_rowsClient<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one Contacts_contact_rows.
   * @param {contacts_contact_rowsUpdateArgs} args - Arguments to update one Contacts_contact_rows.
   * @example
   * // Update one Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends contacts_contact_rowsUpdateArgs>(args: Prisma.SelectSubset<T, contacts_contact_rowsUpdateArgs<ExtArgs>>): Prisma.Prisma__contacts_contact_rowsClient<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more Contacts_contact_rows.
   * @param {contacts_contact_rowsDeleteManyArgs} args - Arguments to filter Contacts_contact_rows to delete.
   * @example
   * // Delete a few Contacts_contact_rows
   * const { count } = await prisma.contacts_contact_rows.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends contacts_contact_rowsDeleteManyArgs>(args?: Prisma.SelectSubset<T, contacts_contact_rowsDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Contacts_contact_rows.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {contacts_contact_rowsUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends contacts_contact_rowsUpdateManyArgs>(args: Prisma.SelectSubset<T, contacts_contact_rowsUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Contacts_contact_rows and returns the data updated in the database.
   * @param {contacts_contact_rowsUpdateManyAndReturnArgs} args - Arguments to update many Contacts_contact_rows.
   * @example
   * // Update many Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more Contacts_contact_rows and only return the `id`
   * const contacts_contact_rowsWithIdOnly = await prisma.contacts_contact_rows.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends contacts_contact_rowsUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, contacts_contact_rowsUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one Contacts_contact_rows.
   * @param {contacts_contact_rowsUpsertArgs} args - Arguments to update or create a Contacts_contact_rows.
   * @example
   * // Update or create a Contacts_contact_rows
   * const contacts_contact_rows = await prisma.contacts_contact_rows.upsert({
   *   create: {
   *     // ... data to create a Contacts_contact_rows
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the Contacts_contact_rows we want to update
   *   }
   * })
   */
  upsert<T extends contacts_contact_rowsUpsertArgs>(args: Prisma.SelectSubset<T, contacts_contact_rowsUpsertArgs<ExtArgs>>): Prisma.Prisma__contacts_contact_rowsClient<runtime.Types.Result.GetResult<Prisma.$contacts_contact_rowsPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of Contacts_contact_rows.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {contacts_contact_rowsCountArgs} args - Arguments to filter Contacts_contact_rows to count.
   * @example
   * // Count the number of Contacts_contact_rows
   * const count = await prisma.contacts_contact_rows.count({
   *   where: {
   *     // ... the filter for the Contacts_contact_rows we want to count
   *   }
   * })
  **/
  count<T extends contacts_contact_rowsCountArgs>(
    args?: Prisma.Subset<T, contacts_contact_rowsCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], Contacts_contact_rowsCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a Contacts_contact_rows.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {Contacts_contact_rowsAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends Contacts_contact_rowsAggregateArgs>(args: Prisma.Subset<T, Contacts_contact_rowsAggregateArgs>): Prisma.PrismaPromise<GetContacts_contact_rowsAggregateType<T>>

  /**
   * Group by Contacts_contact_rows.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {contacts_contact_rowsGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends contacts_contact_rowsGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: contacts_contact_rowsGroupByArgs['orderBy'] }
      : { orderBy?: contacts_contact_rowsGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, contacts_contact_rowsGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetContacts_contact_rowsGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the contacts_contact_rows model
 */
readonly fields: contacts_contact_rowsFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for contacts_contact_rows.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__contacts_contact_rowsClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  agency<T extends Prisma.contacts_contact_rows$agencyArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.contacts_contact_rows$agencyArgs<ExtArgs>>): Prisma.Prisma__agencies_agency_rowsClient<runtime.Types.Result.GetResult<Prisma.$agencies_agency_rowsPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the contacts_contact_rows model
 */
export interface contacts_contact_rowsFieldRefs {
  readonly id: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly first_name: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly last_name: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly email: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly phone: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly title: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly email_type: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly contact_form_url: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly created_at: Prisma.FieldRef<"contacts_contact_rows", 'DateTime'>
  readonly updated_at: Prisma.FieldRef<"contacts_contact_rows", 'DateTime'>
  readonly department: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly agency_id: Prisma.FieldRef<"contacts_contact_rows", 'String'>
  readonly firm_id: Prisma.FieldRef<"contacts_contact_rows", 'String'>
}
    

// Custom InputTypes
/**
 * contacts_contact_rows findUnique
 */
export type contacts_contact_rowsFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
  /**
   * Filter, which contacts_contact_rows to fetch.
   */
  where: Prisma.contacts_contact_rowsWhereUniqueInput
}

/**
 * contacts_contact_rows findUniqueOrThrow
 */
export type contacts_contact_rowsFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
  /**
   * Filter, which contacts_contact_rows to fetch.
   */
  where: Prisma.contacts_contact_rowsWhereUniqueInput
}

/**
 * contacts_contact_rows findFirst
 */
export type contacts_contact_rowsFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
  /**
   * Filter, which contacts_contact_rows to fetch.
   */
  where?: Prisma.contacts_contact_rowsWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of contacts_contact_rows to fetch.
   */
  orderBy?: Prisma.contacts_contact_rowsOrderByWithRelationInput | Prisma.contacts_contact_rowsOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for contacts_contact_rows.
   */
  cursor?: Prisma.contacts_contact_rowsWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` contacts_contact_rows from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` contacts_contact_rows.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of contacts_contact_rows.
   */
  distinct?: Prisma.Contacts_contact_rowsScalarFieldEnum | Prisma.Contacts_contact_rowsScalarFieldEnum[]
}

/**
 * contacts_contact_rows findFirstOrThrow
 */
export type contacts_contact_rowsFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
  /**
   * Filter, which contacts_contact_rows to fetch.
   */
  where?: Prisma.contacts_contact_rowsWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of contacts_contact_rows to fetch.
   */
  orderBy?: Prisma.contacts_contact_rowsOrderByWithRelationInput | Prisma.contacts_contact_rowsOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for contacts_contact_rows.
   */
  cursor?: Prisma.contacts_contact_rowsWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` contacts_contact_rows from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` contacts_contact_rows.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of contacts_contact_rows.
   */
  distinct?: Prisma.Contacts_contact_rowsScalarFieldEnum | Prisma.Contacts_contact_rowsScalarFieldEnum[]
}

/**
 * contacts_contact_rows findMany
 */
export type contacts_contact_rowsFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
  /**
   * Filter, which contacts_contact_rows to fetch.
   */
  where?: Prisma.contacts_contact_rowsWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of contacts_contact_rows to fetch.
   */
  orderBy?: Prisma.contacts_contact_rowsOrderByWithRelationInput | Prisma.contacts_contact_rowsOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing contacts_contact_rows.
   */
  cursor?: Prisma.contacts_contact_rowsWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` contacts_contact_rows from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` contacts_contact_rows.
   */
  skip?: number
  distinct?: Prisma.Contacts_contact_rowsScalarFieldEnum | Prisma.Contacts_contact_rowsScalarFieldEnum[]
}

/**
 * contacts_contact_rows create
 */
export type contacts_contact_rowsCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
  /**
   * The data needed to create a contacts_contact_rows.
   */
  data?: Prisma.XOR<Prisma.contacts_contact_rowsCreateInput, Prisma.contacts_contact_rowsUncheckedCreateInput>
}

/**
 * contacts_contact_rows createMany
 */
export type contacts_contact_rowsCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many contacts_contact_rows.
   */
  data: Prisma.contacts_contact_rowsCreateManyInput | Prisma.contacts_contact_rowsCreateManyInput[]
  skipDuplicates?: boolean
}

/**
 * contacts_contact_rows createManyAndReturn
 */
export type contacts_contact_rowsCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * The data used to create many contacts_contact_rows.
   */
  data: Prisma.contacts_contact_rowsCreateManyInput | Prisma.contacts_contact_rowsCreateManyInput[]
  skipDuplicates?: boolean
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * contacts_contact_rows update
 */
export type contacts_contact_rowsUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
  /**
   * The data needed to update a contacts_contact_rows.
   */
  data: Prisma.XOR<Prisma.contacts_contact_rowsUpdateInput, Prisma.contacts_contact_rowsUncheckedUpdateInput>
  /**
   * Choose, which contacts_contact_rows to update.
   */
  where: Prisma.contacts_contact_rowsWhereUniqueInput
}

/**
 * contacts_contact_rows updateMany
 */
export type contacts_contact_rowsUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update contacts_contact_rows.
   */
  data: Prisma.XOR<Prisma.contacts_contact_rowsUpdateManyMutationInput, Prisma.contacts_contact_rowsUncheckedUpdateManyInput>
  /**
   * Filter which contacts_contact_rows to update
   */
  where?: Prisma.contacts_contact_rowsWhereInput
  /**
   * Limit how many contacts_contact_rows to update.
   */
  limit?: number
}

/**
 * contacts_contact_rows updateManyAndReturn
 */
export type contacts_contact_rowsUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * The data used to update contacts_contact_rows.
   */
  data: Prisma.XOR<Prisma.contacts_contact_rowsUpdateManyMutationInput, Prisma.contacts_contact_rowsUncheckedUpdateManyInput>
  /**
   * Filter which contacts_contact_rows to update
   */
  where?: Prisma.contacts_contact_rowsWhereInput
  /**
   * Limit how many contacts_contact_rows to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * contacts_contact_rows upsert
 */
export type contacts_contact_rowsUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
  /**
   * The filter to search for the contacts_contact_rows to update in case it exists.
   */
  where: Prisma.contacts_contact_rowsWhereUniqueInput
  /**
   * In case the contacts_contact_rows found by the `where` argument doesn't exist, create a new contacts_contact_rows with this data.
   */
  create: Prisma.XOR<Prisma.contacts_contact_rowsCreateInput, Prisma.contacts_contact_rowsUncheckedCreateInput>
  /**
   * In case the contacts_contact_rows was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.contacts_contact_rowsUpdateInput, Prisma.contacts_contact_rowsUncheckedUpdateInput>
}

/**
 * contacts_contact_rows delete
 */
export type contacts_contact_rowsDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
  /**
   * Filter which contacts_contact_rows to delete.
   */
  where: Prisma.contacts_contact_rowsWhereUniqueInput
}

/**
 * contacts_contact_rows deleteMany
 */
export type contacts_contact_rowsDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which contacts_contact_rows to delete
   */
  where?: Prisma.contacts_contact_rowsWhereInput
  /**
   * Limit how many contacts_contact_rows to delete.
   */
  limit?: number
}

/**
 * contacts_contact_rows.agency
 */
export type contacts_contact_rows$agencyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the agencies_agency_rows
   */
  select?: Prisma.agencies_agency_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the agencies_agency_rows
   */
  omit?: Prisma.agencies_agency_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.agencies_agency_rowsInclude<ExtArgs> | null
  where?: Prisma.agencies_agency_rowsWhereInput
}

/**
 * contacts_contact_rows without action
 */
export type contacts_contact_rowsDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the contacts_contact_rows
   */
  select?: Prisma.contacts_contact_rowsSelect<ExtArgs> | null
  /**
   * Omit specific fields from the contacts_contact_rows
   */
  omit?: Prisma.contacts_contact_rowsOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.contacts_contact_rowsInclude<ExtArgs> | null
}
